astropy/io/ascii/cds.py

```diff
@@ -114,9 +116,15 @@ def get_cols(self, lines):
                 col.start = int(re.sub(r'[-\s]', '',
                                        match.group('start') or match.group('end'))) - 1
                 col.end = int(match.group('end'))
-                col.unit = match.group('units')
-                if col.unit == '---':
+                unit = match.group('units')
+                if unit == '---':
                     col.unit = None  # "---" is the marker for no unit in CDS table
+                else:
+                    try:
+                        col.unit = Unit(unit, format='cds')
```

Comment:
I'm not sure we should fail on the unit not parsing correctly - that forces the user to edit the files they downloaded. Instead, I would suggest to follow the fits example and just add `parse_strict='warn'` or even `parse_strict='silent'` (which means one get `UnrecognizedUnit` after all, but of course only for those units that are, in fact, not recognized).

But best to cc @taldcroft, since he will have a better idea about what the standard is in `io.ascii` (perhaps there is a keyword argument already for it??).